File: 프로그래머스/lv2/42577. 전화번호 목록/전화번호 목록.java
import java.util.*;

class Solution {  
    public boolean solution(String[] phone_book) {
        Set<String> phoneSet = new HashSet<>();
        Arrays.sort(phone_book,(o1,o2) -> o2.length() - o1.length());
        for(String phone : phone_book){
            if(phoneSet.contains(phone)){
                return false;
            }
            StringBuilder sb = new StringBuilder();
            for(int i=0;i<phone.length();i++){
                sb.append(phone.charAt(i));
                phoneSet.add(sb.toString());
            }
        }
        
        return true;
    }

}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    answer = True
    list = set()
    
    phone_book.sort(key= len, reverse=True)

    for phone in phone_book:
        num = ""
        
        if phone in list:
            answer=False
            break
        
        for i in phone:
            num = num + i
            list.add(num)
    
    return answer